lib/PrintAsClang/_SwiftCxxInteroperability.h

```diff
@@ -279,6 +279,89 @@ class Error {
   void * _Nonnull opaqueValue = nullptr;
 };
 
+/// The Expected class has either an error or an value.
+template<class T>
+class Expected {
+public:
+
+  /// Default
+  constexpr Expected() { has_val = false; }
+  constexpr Expected(T buf) : buffer(buf) { has_val = true; }
+  constexpr Expected(swift::Error * _Nonnull error_val)  : err(error_val) { has_val = false; }
+
+  /// Copy
+  constexpr Expected(Expected const& other) noexcept {
+    if (other.has_value()) {
+      buffer = other.buffer;
+      has_val = true;
+    }
+  }
+  /// Move
+  constexpr Expected(Expected&& other) noexcept {
+    if (other.has_value()) {
+      buffer = other.buffer;
+      has_val = true;
+    }
+  }
+
+  ~Expected() noexcept { }
+
+  /// assignment
+  constexpr auto operator=(Expected&& other) noexcept { auto temp = buffer; buffer = other.buffer; other.buffer = temp; }
+  constexpr auto operator=(Expected&) noexcept = delete;
+
+  /// For accessing T's members
+  // precondition: has_value() == true
+  constexpr T const* _Nonnull operator->() const noexcept { return &buffer; }
+
+  // precondition: has_value() == true
+  constexpr T* _Nonnull operator->() noexcept { return &buffer; }
+
+  /// Getting reference to T
+  // precondition: has_value() == true
+  constexpr T const& operator*() const& noexcept { return buffer; }
+
+  // precondition: has_value() == true
+  constexpr T& operator*() & noexcept { return buffer; }
+
+  constexpr explicit operator bool() const noexcept {
+    return has_value();
+  }
+
+  // Get value, if not exists abort
+  // FIXME: throw exception instead of abort?
+  constexpr T const& value() const& {
+    if (!has_value())
+      abort();
+    return *reinterpret_cast<const T *>(buffer);
+  }
+
+  constexpr T& value() & {
+    if (!has_value())
+      abort();
+    return *reinterpret_cast<T *>(buffer);
+  }
+
+  // Get error
+  constexpr swift::Error const& error() const& {
+    if (has_value())
+      abort();
+    return err;
+  }
+  constexpr swift::Error& error() & {
+    if (has_value())
+      abort();
+    return err;
+  }
+
+  constexpr bool has_value() const noexcept { return has_val; }
+
+private:
+  T buffer;
```

Comment:
This does not work. Both `T` and `Error` values can't coexist in this value at the same time.
I suggest storing them in a buffer that's appropriately sized and aligned for both of them, and using placement new and manually dispatched destructor call instead.

Reply:
Done!